src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -264,19 +265,26 @@ def _migrate_service_principals(
                     f"Service principal '{spn.permission_mapping.principal}' accesses storage account "
                     f"'{spn.permission_mapping.prefix}' with non-Allow network configuration"
                 )
+            try:
 
-            storage_credential_info = self._storage_credential_manager.create_with_client_secret(spn)
-            validation_results = self._storage_credential_manager.validate(
-                storage_credential_info,
-                spn.permission_mapping.prefix,
-            )
-            execution_result.append(validation_results)
+                storage_credential_info = self._storage_credential_manager.create_with_client_secret(spn)
+                validation_results = self._storage_credential_manager.validate(
+                    storage_credential_info,
+                    spn.permission_mapping.prefix,
+                )
+                execution_result.append(validation_results)
+            except (PermissionDenied, NotFound, BadRequest) as e:
+                logger.error("Error creating storage credentials")
+                self._resource_permissions.delete_storage_credentials(execution_result)
```

Comment:
So, we require all SP to be migrated at once and do not allow for some to be migrated?

Reply:
In order to be a clean run, my suggestion is to either do all or roll back all. We don't want to leave it in in between state where it created 3 access connector but failed in the 4th one or it created all access connector but failed while creating storage credential for them. do you agree?

Reply:
Yes i have made the changes now.
 so all access connectors and associated storage credentials or roll back everything
migrate all service principal or rollback everything
but if access connectors and associated storage credential complete but migrate service principal fail, we only roll back migrate service principal ones